server/routes/spotify.js

```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const router = express.Router();
+require("dotenv").config();
+const querystring = require("querystring");
+const request = require("request");
+const User = require("../models/user");
+
+let username; // expand scope, get at login, use to add refresh token
```

Comment:
I'm not sure this is thread-safe, and doesn't seem like a good practice anyway. Maybe try passing is as a query parameter together with the redirect_uri (e.g "http://localhost:3001/spotify/callback?user=${username}", then getting it back on the /callback